src/modules/awake/Awake/Program.cs

```diff
@@ -207,22 +237,19 @@ private static void HandleCommandLineArguments(bool usePtConfig, bool displayOn,
                 }
             }
 
-            var exitSignal = new ManualResetEvent(false);
             if (pid != 0)
             {
                 RunnerHelper.WaitForPowerToysRunner(pid, () =>
                 {
-                    exitSignal.Set();
-                    Environment.Exit(0);
+                    Exit("Terminating from PowerToys binding hook.", 0, true);
                 });
             }
 
-            exitSignal.WaitOne();
+            _exitSignal.WaitOne();
```

Comment:
The event is never signaled. Do we really need it here?

Reply:
This is signaled - try running the application in standalone mode. This line ensures that the process is up and running without the app considering that it was done with execution.

Reply:
> This is signaled

It is not signaled by our code. It looks like the only purpose of this event is to keep the process alive. Is it correct?